src/HubSpot/LineItems/LineItem.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace HubSpot.LineItems
+{
+    public class LineItem : IHubSpotEntity
+    {
+        [DefaultProperty("hs_object_id")]
+        public long Id { get; set; }
+
+        [CustomProperty("createdate")]
+        public DateTime CreateDate { get; set; }
+
+        [CustomProperty("hs_lastmodifieddate")]
+        public DateTime LastUpdated { get; set; }
+
+        [CustomProperty("name")]
+        public string Name { get; set; }
+
+        [CustomProperty("hs_sku")]
+        public string SKU { get; set; }
+
+        [CustomProperty("hs_term_in_months")]
+        public int MonthsTerm { get; set; }  
+        
+        [CustomProperty("hs_recurring_billing_start_date")]
+        public DateTime StartDate { get; set; }
+
+        [CustomProperty("quantity")]
+        public int Quantity { get; set; }
+
+        [CustomProperty("price")]
+        public decimal Price { get; set; }
+
+        [CustomProperty("discount")]
+        public decimal Discount { get; set; }
+
+        [CustomProperty("amount")]
+        public decimal NetPrice { get; set; }
+
+        [CustomProperty("hs_margin_mrr")]
+        public decimal MRR { get; set; }  
+        
+        [CustomProperty("hs_arr")]
+        public decimal ARR { get; set; }   
+        
+        [CustomProperty("hs_margin_tcv")]
+        public decimal TCV { get; set; }
+
+        [CustomProperty("number_of_services")]
+        public int NumberOfServices { get; set; }  
+        
+        [CustomProperty("contract_start_date")]
+        public DateTime? ContractStartDate { get; set; }    
+        
+        [CustomProperty("contract_expiration_date")]
+        public DateTime? ContractEndDate { get; set; }  
+        
+        [CustomProperty("service_start_date")]
+        public DateTime ServiceStartDate { get; set; }      
+        
+        [CustomProperty("service_end_date")]
+        public DateTime ServiceEndDate { get; set; }
+
```

Comment:
Are these custom properties? In that case we might want to make them nullable in the sdk, to avoid issues if they are removed (if we even want to have them hardcoded in the sdk).
For the EMG custom properties we don't have them hardcoded in the sdk, we have it in the hubspot RA, can that be done for these as well?

Reply:
The **CustomProperty**  attribute was misleading to me and it took me some time to figure it out looking at the other crm objects we retrieve, but it does not necessarily mean that a property must have been custom created in hubspot, what it sort of implies is that said property will be available inside the "properties" dictionary of the entity we retrieve from hubspot (link [here](https://legacydocs.hubspot.com/docs/methods/deals/get_deal) for an example of what I mean).
So in this case those properties are not made by us, they exist in the hubspot default line item, which is why I haven't added them in the RA.
As a note, starting with v3 of hubspot api (or maybe its like this in v2 as well, but v1 I checked and its not like this), if you want to retrieve data about a crm object, you need to make a query with the list of all columns you want retrieved, otherwise you will only get only 2-3 columns (id and archived I think), so this is why I added more columns than we will actually use for this ticket.